synctool/src/main/java/org/duracloud/sync/endpoint/DuraStoreChunkSyncEndpoint.java

```diff
@@ -193,16 +198,81 @@ private ChunksManifest getManifest(String spaceId, String contentId) {
     protected void addUpdateContent(String contentId,
                                     MonitoredFile syncFile) {
         Map<String, String> properties = createProps(syncFile.getAbsolutePath(), getUsername());
+        final ContentStore store = getContentStore();
+
         DuracloudContentWriter contentWriter =
-            new DuracloudContentWriter(getContentStore(), getUsername(), true, this.jumpStart);
+            new DuracloudContentWriter(store, getUsername(), true, this.jumpStart);
         FileChunker chunker = new FileChunker(contentWriter, chunkerOptions);
-
-        chunker.addContent(getSpaceId(),
+        final String spaceId = getSpaceId();
+        chunker.addContent(spaceId,
                            contentId,
                            syncFile.getChecksum(),
                            syncFile.length(),
                            syncFile.getStream(),
                            properties);
+
+        //clean up any orphaned chunks and / or obsolete unchunked files
+        try {
+            log.debug("checking for chunks, manifests, and/or unchunked files that should be removed : for {}/{}",
+                      spaceId, contentId);
+            Iterator<String> chunkedContentIdIt = store.getSpaceContents(getSpaceId(), contentId + ".dura-");
+            Set<String> chunkedContentIds = new HashSet<>();
+            chunkedContentIdIt.forEachRemaining(id -> chunkedContentIds.add(id));
+            //if there are chunks
+            if (!chunkedContentIds.isEmpty()) {
+                //clean up
+                if (syncFile.length() <= chunkerOptions.getMaxChunkSize()) {
+                    log.info("A chunked version was replaced by an unchunked version of {}/{}",
+                              spaceId, contentId);
+
+                    //if file is less than or equal to max chunk size
+                    //then we can expect the new file is unchunked
+                    //look for any chunked content matching path
+                    //delete all.
+                    chunkedContentIds.stream().forEach(content -> {
+                        deleteContent(spaceId, content, store);
+                    });
+                    log.info("Deleted manifest and all chunks associated with {}/{}",
+                              spaceId, contentId);
+                } else {
+                    log.info("Checking for orphaned chunks associated with {}/{}",
```

Comment:
This log statement probably be debug level, as there's not really a need for a "checking" log notification on every file.

Reply:
Agreed. Will change.